File: backend/Core/Models/Mappers/CollectionMapper.cs
using Data.Records;

namespace Core.Models.Mappers;

public static class CollectionMapper
{
    public static CollectionModel Map(CollectionRecord collection)
    {
        return new CollectionModel
        {
            Reference = collection.Reference,
            CreatedAt = collection.CreatedAt,
            Name = collection.Name
        };
    }
}

File: frontend/src/api/_shared/mapper/collection.mapper.ts
import dayjs from 'dayjs';

import { IApiCollection } from '@/api/_shared/ApiCollection.type';
import { ICollection } from '@/model/Collection.model';

export const collectionMapper = {

    map(collection: IApiCollection): ICollection {
        return {
            reference: collection.reference,
            createdAt: dayjs(collection.createdAt),
            name: collection.name,
        };
    },

};